drivers/char/broadcom/rpivid-mem.c

```diff
@@ -197,22 +197,29 @@ static int rpivid_mem_probe(struct platform_device *pdev)
 	/* Legacy alias */
 	{
 		char *oldname = kstrdup(priv->name, GFP_KERNEL);
+		struct device *old_mem_dev;
 
 		oldname[1] = 'a';
 		oldname[2] = 'r';
 		oldname[3] = 'g';
 		oldname[4] = 'o';
 		oldname[5] = 'n';
-		(void)device_create(priv->class, NULL, priv->devid + 1, NULL,
-				       oldname + 1);
+		old_mem_dev = device_create(priv->class, NULL, priv->devid + 1,
+					    NULL, oldname + 1);
 		kfree(oldname);
+
+		ptr_err = old_mem_dev;
+		if (IS_ERR(old_mem_dev))
```

Comment:
IS_ERR(ptr_err) would seem more logical here. Or just drop the old_mem_dev and assign directly to ptr_err.

Reply:
I'd copied and pasted from the other device_create failure case. Having ptr_err seems redundant anyway, and the error case may as well do ```err = PTR_ERR(rpivid_mem_dev)``` or ```err = PTR_ERR(rpivid_mem_dev)```.

Actually the handling of ```dev``` as being ```&pdev->dev```, and is then assigned to ```priv-dev``` leaves ```dev``` unused through the rest of the function. I'll do a few more lines of cleanup.